data_processing/transform.py

```diff
@@ -743,6 +744,105 @@ def load_category_and_sub_category():
+        if not pd.isna(record.subcategory):
+            df.at[ind, 'category.id'] = convert_to_url_string(record.category + record.subcategory)
+            df.at[ind, 'category.parent_id'] = convert_to_url_string(record.category)
+            df.at[ind, 'category.name'] = record.subcategory
+        elif not pd.isna(record.category):
+            df.at[ind, 'category.id'] = convert_to_url_string(record.category)
+            df.at[ind, 'category.parent_id'] = None
+            df.at[ind, 'category.name'] = record.category
+    '''
+    The program_to_category table is generated by using the id of each program, and
+    mapping it to the id to each sub_category it is associated with. the category_type
+    of program_to_category is going to be column H fo the csv file.
+    '''
+    # rename columns for easier readability/comprehension
+    df = df.rename(columns = {'type':'program_to_category.category_type', '`': 'program.id', 'name': 'program.name'})
+    # Column C maps to objective of program table
+    df['program.objective'] = df['description'].apply(lambda x: x if not pd.isna(x) else None)
+     # INSERT INTO category table
+    for ind, record in df[df['program.id'].notnull()].drop_duplicates(['category.id']).iterrows():
+        # build query to INSERT record into 'category' table
+        category_query = f"INSERT INTO category (id, type, name, parent_id) VALUES (?, ?, ?, ?);"
+        category_values = tuple([record['category.id'], record['category.type'], record['category.name'], record['category.parent_id']])
+        try:
+            cur.execute(category_query, category_values)
+        except Exception as e:
+            print(str(e))
+            print(f"ERROR - {ind}\n{category_query} ")
+    # INSERT INTO program and program_to_category tables
+    for ind, record in df[df['program.id'].notnull()].iterrows():
+        # build query to INSERT record into 'program' table
+        program_query = f"""INSERT INTO program 
+        (id, agency_id, name, popular_name, objective, sam_url, usaspending_awards_hash, usaspending_awards_url, grants_url, program_type) 
+        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?);"""
+        program_values = tuple([record['program.id'], int(record['program.agency_id']), record['program.name'], None, record['program.objective'], None, None, None, None, record['program_to_category.category_type']])
+        try:
+            cur.execute(program_query, program_values)
+        except Exception as e:
+            # log error...
+            print(str(e))
+            print(f"ERROR - {ind}\n{program_query} ")
+        # build query to ISNERT record into 'program_to_category' table
+        program_to_category_query = f"""INSERT INTO program_to_category (program_id, category_id, category_type) VALUES (?, ?, ?);"""
+        program_to_category_values = tuple([record['program.id'], record['category.id'], record['program_to_category.category_type']])
```

Comment:
This line is on me. Let's set the "category_type" to just `category`. I know that I mentioned we wanted to get the value of column H but I was wrong on that. 

```suggestion
        program_to_category_values = tuple([record['program.id'], record['category.id'], 'category'])
```